build-system/pr-check/experiment-tests.js

```diff
@@ -20,43 +20,40 @@
  */
 
 const experimentsConfig = require('../global-configs/experiments-config.json');
+const {buildTargetsInclude, Targets} = require('./build-targets');
 const {experiment} = require('minimist')(process.argv.slice(2));
 const {printSkipMessage, timedExecOrDie} = require('./utils');
 const {runCiJob} = require('./ci-job');
 
 const jobName = `${experiment}-tests.js`;
 
+/**
+ * Extracts and validates the experiment config.
+ * @return {Object|null}
+ */
 function getConfig_() {
   const config = experimentsConfig[experiment];
-
-  if (!config || !config.name || !config.define_experiment_constant) {
-    return;
-  }
-
-  if (new Date(config['expiration_date_utc']) < Date.now()) {
-    return;
-  }
-
-  return config;
-}
-
-function build_(config) {
-  const command = `gulp dist --fortesting --define_experiment_constant ${config.define_experiment_constant}`;
-  timedExecOrDie('gulp clean');
-  timedExecOrDie('gulp update-packages');
-  timedExecOrDie(command);
-}
-
-function test_() {
-  timedExecOrDie('gulp integration --nobuild --compiled --headless');
-  timedExecOrDie('gulp e2e --nobuild --compiled --headless');
+  const valid =
+    config?.name &&
+    config?.define_experiment_constant &&
+    config?.expiration_date_utc &&
+    new Date(config.expiration_date_utc) >= Date.now();
+  return valid ? config : null;
 }
 
 function pushBuildWorkflow() {
   const config = getConfig_();
   if (config) {
-    build_(config);
-    test_();
+    const defineFlag = `--define_experiment_constant ${config.define_experiment_constant}`;
+    const experimentFlag = `--experiment ${experiment}`;
+    timedExecOrDie('gulp update-packages');
+    timedExecOrDie(`gulp dist --fortesting ${defineFlag}`);
+    timedExecOrDie(
+      `gulp integration --nobuild --compiled --headless ${experimentFlag} ${defineFlag}`
```

Comment:
Does it work with `--nobuild` because you're running `gulp dist` above?

Reply:
Yes, it works. `--nobuild` prevents the test task from rebuilding code all over again.